engine/src/main/java/org/hibernate/search/backend/impl/DeletionQueryMapper.java

```diff
@@ -0,0 +1,38 @@
+/*
+ * Hibernate Search, full-text search for your domain model
+ *
+ * License: GNU Lesser General Public License (LGPL), version 2.1 or later
+ * See the lgpl.txt file in the root directory or <http://www.gnu.org/licenses/lgpl-2.1.html>.
+ */
+package org.hibernate.search.backend.impl;
+
+import org.apache.lucene.search.Query;
+import org.hibernate.search.backend.DeletionQuery;
+import org.hibernate.search.util.impl.ScopedAnalyzer;
+
+/**
+ * @author Martin Braun
+ */
+public interface DeletionQueryMapper {
```

Comment:
Okay. In this case several implementations are possible. I will add a comment.

In my experience, by declaring interfaces first and hiding the logic completely, you are thinking more modular, even if you are not using mock-ups.